design/secrets_provider_milestone1_2_design.md

```diff
@@ -0,0 +1,367 @@
+    + [Flow](#flow)
+    + [Customer Experience](#customer-experience)
+    + [Code changes](#code-changes)
+    + [Fetching Relevant K8s Secrets](#fetching-relevant-k8s-secrets)
+      - [**Customer Experience**](#--customer-experience--)
+  * [Milestone 2: Update K8s secret with Conjur secret](#milestone-2--update-k8s-secret-with-conjur-secret)
+    + [Design](#design-1)
+    + [Flow](#flow-1)
+    + [Customer experience](#customer-experience)
+    + [Code changes](#code-changes-1)
+  * [Order of Deployment](#order-of-deployment)
+      - [Milestone 1 and 2](#milestone-1-and-2)
+  * [Backwards compatibility](#backwards-compatibility)
+      - [Milestone 1 and 2](#milestone-1-and-2-1)
+  * [Performance](#performance)
+  * [Affected Components](#affected-components)
+      - [Milestone 1](#milestone-1)
+      - [Milestone 2](#milestone-2)
+- [Security](#security)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Open questions](#open-questions)
+- [Implementation plan](#implementation-plan)
+
+
+
+| Name                | Link                                                         |
+| ------------------- | ------------------------------------------------------------ |
+| Phase 2 Feature Doc | https://github.com/cyberark/secrets-provider-for-k8s/issues/102 |
+
+## Background
+The Cyberark Secrets Provider for K8s enables customers to use secrets stored and managed in Conjur/DAP vault and consume them as K8s Secrets for their application containers. During Phase 1, an *init* container was deployed in the same Pod as the customer's applications and populate the K8s secrets defined in the customer's manifest. 
+
+### Motivation
+
+The current implementation makes the deployment of the Secrets Provider does not support secret rotation and makes the deployment of the Secrets Provider deeply coupled to the customer's applications which not all customers want.
+
+### Requirement
+
+The solution should support the following capabilities:
+
+1. Deployment, deploy as separate Conjur entity; serve K8s Secrets to many Applications
+2. Rotation, update K8s Secret with Conjur secret values (on trigger or on a timely, configurable basis)
+3. Support removal of deployment
+
+## Solution
+This solution will be divided into milestones.
+
+- Milestone 1: Deployment, deploy as as separate entity
+- Milestone 2: Handle rotation, updating K8s Secrets with Conjur secret value (configurable amount of time or on a trigger)
+
+
+
+### Milestone 1: Deployment
+
+#### Design
+
+The Secrets Provider for K8s will no longer run as an *init* container attached to the application pod, rather run as a separate single Job in a namespace. A Job creates a Pod to perform a task and terminates upon completion. Once the Pod is spun up, it will authenticate to Conjur/DAP in the same way it does today, via authn-k8s. The Pod will then fetch the K8s secret denoted with a label and update them with the Conjur secrets they require.
```

Comment:
@oburstein-hub I explained it under the pros/cons table. Specifically it is a waste of a customers require for a pod to stay up in their environment if it only needed to perform an initial task. Is that not enough?

> **Decision:** Run as a Job
> We decided to run as a Job instead of a Pod because for Milestone 1 we are concerned with delivering Conjur values only once. At this time we are not supporting updating K8s Secrets with Conjur secret across multiple runs so deploying a Pod that will run continuously in the environment of a customer is a waste of their resources.